streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -87,7 +112,7 @@ private JoinWindows(final long beforeMs, final long afterMs) {
      * @throws IllegalArgumentException if {@code timeDifferenceMs} is negative
      */
     public static JoinWindows of(final long timeDifferenceMs) throws IllegalArgumentException {
-        return new JoinWindows(timeDifferenceMs, timeDifferenceMs);
+        return new JoinWindows(timeDifferenceMs, timeDifferenceMs, null, DEFAULT_RETENTION_MS);
```

Comment:
can you document/comment why grace is set to null and segments are not passed here ?

Reply:
@mjsax @tedyu , thanks for the reviews.

@kamalcph , Good catches! I've addressed them.